ldaptive/src/test/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDaoTest.java

```diff
@@ -32,4 +55,26 @@ public void testCalculatePageSize() {
         assertThat(dao.calculatePageSize(5, true), is(5));
         assertThat(dao.calculatePageSize(10, true), is(5 + 1));
     }
+
+    @Test
+    public void testStreamSearchRequestStreamClosesConnection() throws Exception {
+        when(connection.isOpen()).thenReturn(true);
+
+        try (Stream<LdapEntry> ignored = dao.streamSearchRequest(REQUEST, 1)) {
+            verify(connection).open();
+            verify(connection, never()).close();
+        }
+        verify(connection).close();
+    }
+
+    @Test
+    public void testStreamSearchRequestCantOpenConnection() throws Exception {
+        when(connection.open()).thenThrow(LdapException.class);
+
+        try {
+            dao.streamSearchRequest(REQUEST, 1);
+            fail();
+        } catch (LdaptiveDaoException expected) {}
+        verify(connection).open();
+    }
```

Comment:
The only thing it tests is that we are wrapping the checked `LdapException` to the unchecked `LdaptiveDaoException` when it fails to open a connection. since it failed to open the connection close wouldn't be called.